tutorials/5-docker-intro/index.md

```diff
@@ -0,0 +1,290 @@
+  title: Docker Introduction
+  description: Get started with Docker
+  number: 
+tags:
+  - tutorial
+  - dev-ops
+order: 0
+title: Docker introduction
+---
+
+<div class="tm-overline tm-rf-1 tm-lh-title tm-medium tm-muted">Know your way around Docker</div>
+<h1 class="mt-4 mb-6">Docker Introduction</h1>
+
+[Docker](https://www.docker.com) is a kind of computer virtualization that provides valuable performance benefits.
+
+## Concepts
+
+Being different from classic forms of virtualization, Docker introduces new concepts with their own vocabulary.
+
+### Containers
+
+You know how a **computer** operates: it uses a disk and memory to run an operating system and the programs within it.
+
+You know how a **virtual machine** works: its computer host offers it a _virtual_ disk and memory, and the virtual machine uses both to run its own operating system and programs. Virtualization mimics a full computer, which requires disk space and memory. It also takes time to start because the virtualized operating system needs to start from scratch.
+
+Docker uses a feature of Linux called [_namespaces_](https://docs.docker.com/get-started/overview/#the-underlying-technology) and _control groups_. With this feature, Linux can launch programs (even low-level ones of the operating system) in an environment where the programs believe they are in a complete Linux environment on their own, i.e. sandboxed. This is also called containerization. A **container** does not have full disk or full memory access to its host (the Linux environment that started it), only to a subset of it. A Linux system can launch multiple containers at a time.
+
+As opposed to proper virtualization, containerization is fast:
+
+* You do not need to start a whole operating system, only to isolate a new container and run its own programs. You now count the start-up time in seconds instead of minutes.
+* The memory used by the container is only that of the programs it runs while benefiting from the rest of the Linux system in an isolated way.
+
+### Images
+
+Instead of a virtual disk with a full operating system, a container starts from an **image**. The image contains the files that are specific to the container and different from its Linux host. Think of tracing paper recording only the differences laid on top of the host's file system. These can be very small.
+
+You can either create images locally via a **`Dockerfile`** containing the commands that create the image, or you can use an existing image. There are registries that store and/or reference them, like your regular package manager. [Docker Hub](https://hub.docker.com) is the main one, and your local Docker knows how to download directly from it.
+
+This introduces an interesting concept whereby [a machine is described as a file](https://twitter.com/FrancescoCiull4/status/1509458241524224005). This is very useful for reproducibility and DevOps. Further, the images are optimized so that each image is a _diff_ of a parent one.
```

Comment:
```suggestion
This introduces an interesting concept whereby [a machine is described as a file](https://nitter.net/FrancescoCiull4/status/1509458241524224005). This is very useful for reproducibility and DevOps. Further, the images are optimized so that each image is a _diff_ of a parent one.
```

:trollface:

Reply:
What's the advantage of nitter?